pkg/model/issuerdiscovery.go

```diff
@@ -51,6 +53,8 @@ type oidcDiscovery struct {
 }
 
 func (b *IssuerDiscoveryModelBuilder) Build(c *fi.ModelBuilderContext) error {
+	ctx := context.TODO()
```

Comment:
Wonder if we should start propagating context through `fi.Context` or something here.

Reply:
I did split out "just this PR" in https://github.com/kubernetes/kops/pull/14778, and I tried adding the context to every callsite of vfs.Context in https://github.com/kubernetes/kops/pull/14777.  Adding to every callsite is pretty overwhelming.

I like the idea of basically extending context.Context very much.  That feels very much within the spirit of the rules.  I might give that a go later on :-)